extension/src/popup/helpers/soroban.ts

```diff
@@ -109,10 +115,33 @@ export const getTokenBalance = (
   );
 };
 
+export const getContractDecimals = async (
```

Comment:
similar to the code in `getTokenBalances`, but broken out here

Reply:
I don't have a suggested change here but one thought - we end up having these very similar helpers to get contract data, but the only difference between them is - the contract ID, the function name, the args to the function, and the decoders used on the response. I think there's probably some abstraction here, and even cooler I wonder how feasible it would be to write something that can take the contract function signature as input and give you everything else we need for this as output. This looks good to me for this though.